vdoctl/cmd/drivers.go

```diff
@@ -73,13 +75,28 @@ var driversCmd = &cobra.Command{
 		// Check the vdoDeployment Namespace and confirm if VDO operator is running in the env
 		getVdoNamespace(ctx)
 
-		err, _ := IsVDODeployed(ctx)
+		var isConfigRequired bool
+		configKey := types.NamespacedName{
+			Namespace: VdoCurrentNamespace,
+			Name:      CompatMatrixConfigMap,
+		}
+		cm := v1.ConfigMap{}
+
+		_ = K8sClient.Get(ctx, configKey, &cm)
+		if len(cm.Data) == 0 {
+			isConfigRequired = true
+			color.Yellow("Since you have not configured the Compatibility Matrix so the default values of the matrix will be taken, If you want to configure the matrix manually then you can skip this step and run 'vdoctl configure compatibility matrix")
+		}
+
+		err, deployment := IsVDODeployed(ctx)
 		if err != nil {
 			if apierrors.IsNotFound(err) {
 				fmt.Println(VDO_NOT_DEPLOYED)
 				return
 			} else {
-				cobra.CheckErr(err)
+				if !isConfigRequired {
```

Comment:
why do we need this check here

Reply:
Since we have not configured compat matrix,
So the pod will be in `createConfigError` state, and isVdoDeployed will throw error not enough replicas of VDO, we are checking if compat matrix is not configured, ignore the error.

Reply:
why not move code in line 136 that creates the default compat-matrix to line 91 before we check if vdo is deployed

Reply:
we need to think about a scenario when VDO is not deployed then we won't be able to get the namespace in which we need to check for compat-matrix

Reply:
in the case of the operator, compatiblity-matrix is a pre-requisite. we should be addressing that before attempting deploy of the operator. it doesnt make sense to bring up the operator in error state when you have control on wiring in the pre-requisite